app/web_framework/cluster_master_application.py

```diff
@@ -253,3 +255,15 @@ def get(self):
         self.write({
             'events': analytics.get_events(since_timestamp, since_id),
         })
+
+
+class _SlaveShutdownHandler(_ClusterMasterBaseHandler):
+    @authenticated
+    def post(self):
+        shutdown_all = self.decoded_body.get('shutdown_all')
+        slaves_to_shutdown = self._cluster_master.all_slaves_by_id().keys() if shutdown_all else\
+            [int(slave_id) for slave_id in self.decoded_body.get('slaves')]
+
+        for slave_id in slaves_to_shutdown:
+            slave = self._cluster_master.get_slave(slave_id)
```

Comment:
Should we assemble the entire list of slaves first before doing shutdown on any of them? If the user has a single invalid slave id halfway through the list, only the first half will be shutdown, and then this will return a 404 error.

Also/alternatively, should we return a response with the list of successfully shutdown slaves?